File: hackerrank/problem solving/algorithms/1 - warmup/7 - staircase/with morfo library/staircase.ts
// https://www.hackerrank.com/challenges/staircase/problem?isFullScreen=true

import { read } from 'morfo';

const n: number = read(Number);
staircase(n);

// T: O(n^2)
// S: O(n) extra space
function staircase(n: number) {
  for (let i = 1; i <= n; i++) {
    const spaces = ' '.repeat(n - i);
    const hashes = '#'.repeat(i);
    console.log(spaces + hashes);
  }
}

File: hackerrank/problem solving/algorithms/1 - warmup/7 - staircase/with morfo library/staircase.go
// https://www.hackerrank.com/challenges/staircase/problem?isFullScreen=true

package main

import (
    "fmt"
    "strings"
    "morfo/morfoio"
)

func main() {
    n := morfoio.Read[int]()
    staircase(n)
}

// T: O(n^2)
// S: O(n) extra space
func staircase(n int) {
    for i := 1; i <= n; i++ {
        spaces := strings.Repeat(" ", n-i)
        hashes := strings.Repeat("#", i)
        fmt.Println(spaces + hashes)
    }
}
